ludwig/data/preprocessing.py

```diff
@@ -1661,6 +1669,62 @@ def _handle_missing_values(
         raise ValueError(f"Invalid missing value strategy {missing_value_strategy}")
 
 
+def handle_llm_prompt_injection(
+    dataset_cols, feature_configs, feature_name_to_preprocessing_parameters, backend
+) -> None:
+    """If output feature is a category feature and it's preprocessing has prompt_template, then we need to inject
+    the prompt into the input text feature(s)."""
+
+    # Recover input and output features
+    input_features = []
```

Comment:
Ideally yes, although propagating model type down to this point isn't very clear, I've been trying to see how it can get passed down. Might require quite a few changes to get that through, but I do think it's probably necessary.